test/unit/select_plan_test.lua

```diff
@@ -339,3 +339,203 @@ g.test_table_count = function()
+        -- Expected values.
+        sharding_key = { 'John', 42 },
+    },
+    extract_table_sharding_key_from_eq_an_ge_conditions_returns_nil = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'name', 'John' }, { '>=', 'age', 42 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_table_sharding_key_from_partial_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }, { fieldno = 4 }} },
+        conditions = {{ '==', 'full_name', { nil,  'Doe' } }, { '==', 'id', 1 }},
+        -- Expected values.
+        sharding_key = { 1, 'Doe' },
+    },
+    extract_table_sharding_key_from_partial_conditions_with_nil_and_non_nil_for_same_value_in_different_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'full_name', { 'John',  'Doe' } }, { '==', 'name_id', { nil, 1 } }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+}
+
+for name, case in pairs(extract_sharding_key_from_conditions_cases) do
+    g[('test_%s'):format(name)] = function()
+        local space_indexes = box.space.customers.index
+        local space_format = box.space.customers:format()
+        local fieldno_map = utils.get_format_fieldno_map(space_format)
+        local conditions = compare_conditions.parse(case.conditions)
+
+        local sharding_key = select_plan.internal.extract_sharding_key_from_conditions(conditions, case.sharding_index,
+                                                                                       space_indexes, fieldno_map)
+        t.assert_equals(sharding_key, case.sharding_key)
+    end
+end
+
+g.before_test('test_extract_sharding_key_from_conditions_for_index_and_field_with_same_name', function()
+    box.space.customers:create_index('city', {
+        parts = { {field = 'id'}, {field = 'city'} },
+        unique = false,
+        if_not_exists = true,
+    })
+end)
+
+g.after_test('test_extract_sharding_key_from_conditions_for_index_and_field_with_same_name', function()
+    box.space.customers.index.city:drop()
+end)
+
+g.test_extract_sharding_key_from_conditions_for_index_and_field_with_same_name = function()
+    local space_indexes = box.space.customers.index
+    local space_format = box.space.customers:format()
+    local fieldno_map = utils.get_format_fieldno_map(space_format)
+
+    local sharding_index = { parts = {{ fieldno = 6 }} }
+
+    local conditions = compare_conditions.parse({{ '==', 'city', { 1, 'New York' } }})
+    local sharding_key = select_plan.internal.extract_sharding_key_from_conditions(conditions, sharding_index,
+                                                                                   space_indexes, fieldno_map)
+    t.assert_equals(sharding_key, { 'New York' }, "Extracted sharding key from index in case of name collision")
```

Comment:
Does it look better or not?

```lua
--- a/test/unit/select_plan_test.lua
+++ b/test/unit/select_plan_test.lua
@@ -535,7 +535,9 @@ g.test_extract_sharding_key_from_conditions_for_index_and_field_with_same_name =
     local sharding_index = { parts = {{ fieldno = 6 }} }
 
     local conditions = compare_conditions.parse({{ '==', 'city', { 1, 'New York' } }})
-    local sharding_key = select_plan.internal.extract_sharding_key_from_conditions(conditions, sharding_index,
-                                                                                   space_indexes, fieldno_map)
-    t.assert_equals(sharding_key, { 'New York' }, "Extracted sharding key from index in case of name collision")
+    local extract_sharding_key = select_plan.internal.extract_sharding_key_from_conditions
+    local sharding_key = extract_sharding_key(conditions, sharding_index,
+                                              space_indexes, fieldno_map)
+    t.assert_equals(sharding_key, { 'New York' },
+        "Extracted sharding key from index in case of name collision")
 end
```

Reply:
Used a shortcut and extra line break